gevent_async/executor.py

```diff
@@ -0,0 +1,41 @@
+import functools
+from concurrent.futures import Future
+from typing import Callable, TypeVar
+
+from gevent import threadpool
+from typing_extensions import ParamSpec
+
+T = TypeVar("T")
+P = ParamSpec("P")
+
+
+class GeventExecutor(threadpool.ThreadPoolExecutor):
```

Comment:
Hi @cretz, I believe gevent.threadpoo.ThreadPoolExecutor is the *unpatched* thread pool, so really this PR is just using native threads.

cf. https://www.gevent.org/api/gevent.threadpool.html#gevent.threadpool.ThreadPoolExecutor :

> A version of [concurrent.futures.ThreadPoolExecutor](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.ThreadPoolExecutor) that always uses native threads, even when threading is monkey-patched.

I verified by running this code locally with the patched `concurrent.futures.ThreadPoolExecutor` and it throws this error, as seen before in our explorations:

```
[~/temporalio_gevent/src] ∴ ~/.local/bin/poetry run python -m gevent_async.worker
INFO:root:Worker started, ctrl+c to exit
INFO:temporalio.workflow:Running workflow with parameter Temporal ({'attempt': 1, 'namespace': 'default', 'run_id': '543fef67-e25d-4bd8-926a-cca7bc78fd91', 'task_queue': 'gevent_async-task-queue', 'workflow_id': 'gevent_async-workflow-id', 'workflow_type': 'GreetingWorkflow'})
ERROR:temporalio.worker._workflow:Failed handling activation on workflow with run ID 543fef67-e25d-4bd8-926a-cca7bc78fd91
Traceback (most recent call last):
  File "/home/discord/.cache/pypoetry/virtualenvs/temporalio-gevent-EEP-UPYz-py3.10/lib/python3.10/site-packages/temporalio/worker/_workflow.py", line 208, in _handle_activation
    completion = await asyncio.wait_for(
  File "/usr/lib/python3.10/asyncio/tasks.py", line 405, in wait_for
    loop = events.get_running_loop()
RuntimeError: no running event loop
```

Reply:
Exactly. We are _intentionally_ using native threads so that gevent doesn't try to mess with them so you can have asyncio running in a native thread and gevent doing other things elsewhere. The whole crux of the original issue is that you couldn't even use Temporal and gevent in the same process because gevent tries to take over, but this allows them to share the process space.

If we need further investigation on using greenlets maybe from activities or something, we can look into that separately.

Reply:
@cretz please correct me if my understanding is wrong: if this is intentionally using native threads, it is creating a new thread for each activity it runs [on line 39](https://github.com/temporalio/samples-python/pull/84/files#diff-e15a336a556fdd3034e734ee18a80f0d12180da39ac407b655174ff085eef8e9R39) despite the comment on line 38 saying that it will "[s]ubmit our wrapper to gevent". this means that all the code executed in activities needs to be threadsafe.

our codebase isn't threadsafe (pretty common for gevent codebases i think) but we are still giving this a go to see if we can get basic execution working.